services/server/test/integration/verification-handlers/verify.stateless.spec.ts

```diff
@@ -335,13 +336,41 @@ describe("/", function () {
 
     const partialMetadataURL = `/repository/contracts/partial_match/${chainFixture.chainId}/${chainFixture.defaultContractAddress}/metadata.json`;
 
+    // Intercept only eth_getTransactionReceipt calls but allow other RPC calls
+    // This will block the binary search for the creation tx hash and creationMatch will be set to null
+    nock("http://localhost:8545")
+      .post("/")
+      .reply(function (uri, requestBody) {
+        const body =
+          typeof requestBody === "string"
+            ? JSON.parse(requestBody)
+            : requestBody;
+        if (body.method === "eth_getTransactionReceipt") {
+          return [
+            500,
+            {
+              error: {
+                message: "Blocked getTransactionReceipt",
+              },
+            },
+          ];
+        }
+        // Remove all interceptors and let this request pass through to the real endpoint
+        nock.cleanAll();
```

Comment:
I think it is not safe to call `nock.cleanAll` here. This will also remove the interceptors set up in `LocalChainFixture` and may interfere with other tests.

Maybe it would be better to mock the `SourcifyChain` function with sinon here?

Reply:
Yeah good point. I first tried using stubs but it wouldn't work because we can't modify the the running server instance's SourcifyChain on the fly, right? Or I'm missing something.

Reply:
I haven't tried but I think it should be possible. You just need to access the instance the server is using and overwrite the related function, like `sourcifyChain.getTx = sandbox.stub()`. See `proxy-contract-util.spec.ts` for example. There we use a sandbox to restore after each test. 
For accessing the specific `SourcifyChain` instance, we might need to modify the `LocalChainFixture`.

Reply:
There you are passing the mocked SourcifyChain instances to the functions `detectAndResolveProxy`, they are unit tests. Current Server() setup does not allow "get"ing the chains and modifying them afaik. 

If you are ok with the final setup after the fix I think this is good enough.

Reply:
I'm fine with it. But there is still a `.only` modifier on the test 

Just for reference, shouldn't it be possible to just access the SourcifyChain by `serverFixture.server.chainRepository.sourcifyChainMap[chainId]`?

Reply:
Ooops good catch.

And turns out it was possible. Have a alook

Reply:
Looks good :)